optuna/pruners/hyperband.py

```diff
@@ -70,20 +78,42 @@ class HyperbandPruner(BasePruner):
     """
 
     def __init__(
-        self, min_resource=1, reduction_factor=3, n_brackets=4, min_early_stopping_rate_low=0
-    ):
-        # type: (int, int, int, int) -> None
-
+        self,
+        max_resource: int = 100,
+        min_resource: int = 1,
+        reduction_factor: int = 3,
+        n_brackets: Optional[int] = None,
+        min_early_stopping_rate_low: int = 0,
+    ) -> None:
+        self._max_resource = max_resource
         self._pruners = []  # type: List[SuccessiveHalvingPruner]
         self._reduction_factor = reduction_factor
         self._resource_budget = 0
-        self._n_brackets = n_brackets
+
+        if n_brackets is None:
+            # In the original paper <<http://www.jmlr.org/papers/volume18/16-558/16-558.pdf>, the
```

Comment:
A huge nit but I happen not to be familiar with these brackets surrounding the URL. Are they stemming from a format/syntax?

Reply:
This is not from a format/syntax. I'm wondering whether we should explain where the logic is specified in the original paper.

Reply:
> I'm wondering whether we should explain where the logic is specified in the original paper. 

Sorry I couldn't understand what you mean here. Do you mean to not mention "Algorithm 1"?

Reply:
> This is not from a format/syntax.

How about removing the brackets?